ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -131,9 +133,41 @@ def wan_local_carriers(self: SharedUtils) -> list:
 
         return list(local_carriers_dict.values())
 
+    def _path_group_preference_to_eos_priority(self, path_group_preference: int | str, context_path: str) -> int:
+        """
+        Convert "preferred" to 1 and "alternate" to 2. Everything else is returned as is.
+
+        Arguments:
+        ----------
+        path_group_preference (str|int): The value of the preference key to be converted. It must be either "preferred", "alternate" or an integer.
+        context_path (str): Input path context for the error message.
+        """
+        if path_group_preference == "preferred":
+            return 1
+        if path_group_preference == "alternate":
+            return 2
+        try:
+            return int(path_group_preference)
```

Comment:
While we are at it we should check that the value is between 1 and 255. The schema is not checking this, since it is a string.

Reply:
Why is it 1-255 here, when in eos cli gen it's 1 - 65535. 

Also do we need to validate it here, since cli gen will validate it ?

Reply:
the limitiation is checked in `eos_cil_config_gen` 

and the range is

```
pf2(config-load-balance-policy-LB-CONTROL-PLANE-PROFILE)#path-group PPP priority ?
  <1-65535>  Priority of this group
```

Reply:
I copied over the range 1-255 from the policy schema. Added the check in this function as well, now.

Reply:
If we support 65535 we should indeed change that all over.

Re validating inputs, we should make sure eos_designs produce good output when possible. And part of that is also validating the inputs. From a user perspective it does not make much sense to run through minutes of build time, just to get an input error at the end.